crates/uv-resolver/src/pubgrub/priority.rs

```diff
@@ -80,6 +82,7 @@ impl PubGrubPriorities {
                 let priority = if urls.get(name).is_some() {
                     PubGrubPriority::DirectUrl(Reverse(next))
                 } else if version.as_singleton().is_some() {
+                    // TODO(charlie): Take local version ranges into account (e.g., `[2.0, 2.0+[max])`).
```

Comment:
I'll fix this separately, but this fix alone does _not_ resolve the bug in the linked issue.